src/XrdCl/XrdClAsyncMsgReader.hh

```diff
@@ -168,6 +169,23 @@ namespace XrdCl
               continue;
             }
             //------------------------------------------------------------------
+            // kXR_status is special as it can have both body and raw data,
+            // handle it separately
+            //------------------------------------------------------------------
+            case ReadMore:
+            {
+              XRootDStatus st = xrdTransport.GetMore( *inmsg, &socket );
+              if( !st.IsOK() || st.code == suRetry )
+                return st;
+              inmsgsize = inmsg->GetCursor();
+
+              //----------------------------------------------------------------
+              // The next step is to finalize the read
+              //----------------------------------------------------------------
+              readstage = ReadDone;
```

Comment:
Can `ReadMore` be required more than once?

Reply:
yes, certainly

Reply:
What I wonder is if setting `readstage = ReadDone` will not prevent `ReadMore` from being called twice in a row if needed.